satellite/metainfo/metainfo.go

```diff
@@ -523,15 +523,22 @@ func (endpoint *Endpoint) filterValidPieces(ctx context.Context, pointer *pb.Poi
 			return Error.New("all pieces needs to have the same size")
 		}
 
-		// we repair when the number of healthy files is less than or equal to the repair threshold
-		// except for the case when the repair and success thresholds are the same (a case usually seen during testing)
+		// We repair when the number of healthy files is less than or equal to the repair threshold
+		// except for the case when the repair and success thresholds are the same (a case usually seen during testing).
 		if int32(len(remotePieces)) <= remote.Redundancy.RepairThreshold && int32(len(remotePieces)) < remote.Redundancy.SuccessThreshold {
 			return Error.New("Number of valid pieces (%d) is less than or equal to the repair threshold (%d)",
 				len(remotePieces),
 				remote.Redundancy.RepairThreshold,
 			)
 		}
 
+		if int32(len(remotePieces)) < remote.Redundancy.SuccessThreshold {
```

Comment:
This is preventing from committing into the pointerdb, the info about pieces that are already uploaded and happening to be less than threshold pieces. So satellite won't make a commit segment info into pointerdb. What about the pieces that are already uploaded onto SNOs and now SNOs will be sending those orders to satellite for payment, which satellite will reject. Is my understanding right that this change will potentially use up SNOs bandwidth and space?

Reply:
Yeah I think you are correct. If `CommitSegment` fails, the nodes that data was uploaded to don't seem to be notified in anyway. But I think garbage collection should help with this in the future. Either way, we do not want the satellite to say that a segment is successful if it was not even uploaded to the correct threshold.